Networking/NetworkingTests/Remote/WCPayRemoteTests.swift

```diff
@@ -53,4 +53,269 @@ final class WCPayRemoteTests: XCTestCase {
+            remote.loadAccount(for: self.sampleSiteID) { result in
+                promise(result)
+            }
+        }
+
+        // Then
+        switch result {
+        case .success(let account):
+            XCTAssertEqual(account.status, .restricted)
+            XCTAssertTrue(account.hasPendingRequirements)
+            XCTAssertFalse(account.hasOverdueRequirements)
+            XCTAssertEqual(account.currentDeadline, Date(timeIntervalSince1970: 1897351200))
+        case .failure(let error):
+            XCTAssertNil(error)
+        }
+    }
+
+    /// Verifies that loadAccount properly handles the restricted - over due response
+    ///
+    func test_loadAccount_properly_handles_restricted_over_due_account() {
+        let remote = WCPayRemote(network: network)
+
+        network.simulateResponse(requestUrlSuffix: "payments/accounts", filename: "wcpay-account-restricted-overdue")
+
+        // When
+        let result: Result<WCPayAccount, Error> = waitFor { promise in
+            remote.loadAccount(for: self.sampleSiteID) { result in
+                promise(result)
+            }
+        }
+
+        // Then
+        switch result {
+        case .success(let account):
+            XCTAssertEqual(account.status, .restricted)
+            XCTAssertFalse(account.hasPendingRequirements)
+            XCTAssertTrue(account.hasOverdueRequirements)
+        case .failure(let error):
+            XCTAssertNil(error)
+        }
+    }
+
+    /// Verifies that loadAccount properly handles an unrecognized status response
+    ///
+    func test_loadAccount_properly_handles_unrecognized_status_account() {
+        let remote = WCPayRemote(network: network)
+
+        network.simulateResponse(requestUrlSuffix: "payments/accounts", filename: "wcpay-account-unknown-status")
+
+        // When
+        let result: Result<WCPayAccount, Error> = waitFor { promise in
+            remote.loadAccount(for: self.sampleSiteID) { result in
+                promise(result)
+            }
+        }
+
+        // Then
+        switch result {
+        case .success(let account):
+            XCTAssertEqual(account.status, .unknown)
+        case .failure(let error):
+            XCTAssertNil(error)
+        }
+    }
+
+    /// Verifies that loadAccount properly handles an error response
+    ///
+    func test_loadAccount_properly_handles_error_response() {
+        let remote = WCPayRemote(network: network)
+
+        network.simulateResponse(requestUrlSuffix: "payments/accounts", filename: "wcpay-account-error")
+
+        // When
+        let result: Result<WCPayAccount, Error> = waitFor { promise in
+            remote.loadAccount(for: self.sampleSiteID) { result in
+                promise(result)
+            }
+        }
+
+        // Then
+        switch result {
+        case .success(let account):
+            XCTAssertNil(account)
+        case .failure(let error):
+            XCTAssertNotNil(error)
```

Comment:
Just noting that the `error` is not exactly what we expect. It's a decoding error instead of the error represented by `wcpay-account-error.json`. 

```
(lldb) po error
▿ DecodingError
  ▿ typeMismatch : 2 elements
    - .0 : Swift.String
    ▿ .1 : Context
      ▿ codingPath : 2 elements
        - 0 : CodingKeys(stringValue: "data", intValue: nil)
        - 1 : CodingKeys(stringValue: "status", intValue: nil)
      - debugDescription : "Expected to decode String but found a number instead."
      - underlyingError : nil
```

I think we're supposed to be using `network.simulateError()` for the mock. But we'd need to make a change in there to simulate the JSON response too. That's best done outside of this PR. 🙂

Reply:
Will fix in https://github.com/woocommerce/woocommerce-ios/issues/3900